packages/components/atoms/f-toggle/src/components/Toggle.vue

```diff
@@ -0,0 +1,223 @@
+<template>
+    <label
+        :for="uniqueId"
+        data-test-id="toggle-component"
+        :class="[$style['c-toggle'],
+                 {
+                     [$style['c-toggle--checked']]: isChecked,
+                     [$style['c-toggle--disabled']]: $attrs.disabled,
+                     [$style['c-toggle--modeRTL']]: isModeRightToLeft
+                 }]">
+        <input
+            :id="uniqueId"
+            role="switch"
+            type="checkbox"
+            v-bind="$attrs"
+            :class="$style['c-toggle-input']"
+            :aria-checked="isChecked"
+            :aria-labelledby="ariaLabelledBy"
+            :aria-describedby="ariaDescribedBy"
+            @change="toggleOption">
+
+        <div :class="$style['c-toggle-control']">
+            <check-icon
+                v-if="isChecked"
+                :class="$style['c-toggle-icon']"
+                aria-hidden />
+        </div>
+    </label>
+</template>
+
+<script>
+import { CheckIcon } from '@justeattakeaway/pie-icons-vue';
+import { v4 as uuid } from 'uuid';
+
+export default {
+    name: 'FToggle',
+
+    components: {
+        CheckIcon
+    },
+
+    inheritAttrs: false,
+
+    props: {
+        isModeRightToLeft: {
```

Comment:
have we agreed on this approach as a team? I personally think we shouldn't be passing an RTL prop to every component a RTL consumer might be using

Reply:
I based it off of what Kevin was doing in modal. I've changed it now to use `dir`